src/main/java/edu/cornell/kfs/coa/businessobject/CuAccountGlobal.java

```diff
@@ -124,11 +124,40 @@ private Account updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccount
 			if(ObjectUtils.isNotNull(this.getIndirectCostRecoveryAccounts()) && this.getIndirectCostRecoveryAccounts().size() > 0){
 				updateIcrAccounts(globalDetail, account.getIndirectCostRecoveryAccounts());
 			}
+
+			if (ObjectUtils.isNotNull(this.accountRestrictedStatusCode)) {
+				updateAccountRestrictedStatusCodeForAccountBeingEdited(this.accountRestrictedStatusCode, account);
+			}
 		}
 
 		return account;
 	}
 
+	/**
+	 * The business rules for the accountRestrictedStatusCode is as follows:
+	 * If the sub-fund of the account listed in the edit list of accounts on
+	 * the global maintenance edoc is one that has a default accountRestrictedStatusCode
+	 * listed, then any change entered in the global account maintenance edoc is ignored.
+	 *
+	 * If the sub-fund of the account listed in the edit list of accounts on
+	 * the global maintenance edoc is one that does NOT have a default accountRestrictedStatusCode
+	 * listed, then any change entered in the global account maintenance edoc overwrites
+	 * the accountRestrictedStatusCode value on that particular account from the edit list.
+	 *
+	 * @param globalAccountMaintenanceDocRestrictedStatusCode
+	 * @param accountBeingEdited
+	 */
+	private void updateAccountRestrictedStatusCodeForAccountBeingEdited(String globalAccountMaintenanceDocRestrictedStatusCode, Account accountBeingEdited)
+	{
+		if (ObjectUtils.isNotNull(accountBeingEdited.getSubFundGroup()) && StringUtils.isNotBlank(accountBeingEdited.getSubFundGroup().getAccountRestrictedStatusCode())) {
```

Comment:
I don't really like having an empty if block just to have a comment; I think negating this condition and just having the logic in the else block (also without the comment) would be better. You have explained the logic in the method comment and the method name. If more explanation is needed, maybe better naming, or extracting some of the condition/logic into appropriate named methods would help.